### ISSUE TITLE
recurrent crash when browsing settings 

### ISSUE THREAD

**Issue by pasnox** [realcopacetic/skin.copacetic#128]
very often when I want to switch a setting section, interface, system,... this leads to Kodi crash.
It restarts by itself so not a big issue, but very annoying.

**Comment by realcopacetic** [comment]
Please can you post a full debug log so I can try and identify the underlying issue.

Instructions for how to do it here: https://kodi.wiki/view/Log_file/Easy

Thanks

**Comment by pepszzi** [comment]
Hi, I have the same issue. 
Here is my log: 
https://paste.kodi.tv/miwadapuza.kodi 

+ I don't wanna stress you out but when do you think the OSD will be fixed? 

Thank you :)

**Comment by realcopacetic** [comment]
what's the issue with the OSD @pepszzi

**Comment by pepszzi** [comment]
Some icons are missing: play, pause, subtitles.

I thought we mean the same with this: 
Still to do:
OSD : (still to update --> Buttons, Settings, Stars, icon_Player)

Got this since a few versions:
![image](https://github.com/realcopacetic/skin.copacetic/assets/99938783/9f58c5fb-61e0-41f0-a9a8-922f4492b6a0)

**Comment by realcopacetic** [comment]
> Some icons are missing: play, pause, subtitles.
> 
> I thought we mean the same with this: 
> Still to do:
> OSD : (still to update --> Buttons, Settings, Stars, icon_Player)
> 
> Got this since a few versions:
> ![image](https://github.com/realcopacetic/skin.copacetic/assets/99938783/9f58c5fb-61e0-41f0-a9a8-922f4492b6a0)

I recently changed some but they should all be there. What version are you on? Perhaps try reinstalling the skin, it seems like something hasn't updated properly if some of the button image are missing

**Comment by pepszzi** [comment]
Strange, I've already tried a couple things. I also deinstalled the entire skin, include the settings and reinstalled it. Same result. 
When I downgrade to 1.1.16 the buttons back. 

I'm using the latest version 1.1.26 and I think this occurred with 1.1.22 or 1.1.23. 
Strangely, it works on my Android device. 
On this device (RPi 4) runs Libreelec, maybe that's the reason? Hope you can investigate further and please let me know if I can help. Should I open a new issue for this? @realcopacetic

![image](https://github.com/realcopacetic/skin.copacetic/assets/99938783/035e889a-5aa4-4e2a-be29-8e7e60906f30)

![image](https://github.com/realcopacetic/skin.copacetic/assets/99938783/2e7913f0-f095-464b-8728-4021f04c2364)

**Comment by pasnox** [comment]
I'm running libreelec as well so I can have a look tonight, which kind of media I should try with?
And yes, open a new issue please.

**Comment by pepszzi** [comment]
@pasnox

Okay, that will be helpful. The media doesn't matter, just bring up the OSD with the buttons. Thank you.

_New issue opened, sorry for crashing this._

**Comment by realcopacetic** [comment]
It's because I changed the case of the filename for some buttons for consistency (e.g. icon_Play_unfocused.png > icon_play_unfocused.png.

But it appears GitHub doesn't pick that up as a change. So it hasn't actually updated the filenames https://github.com/realcopacetic/skin.copacetic/tree/master/media%2Fosd

And Linux systems are sensitive to this. Maybe it didn't pick it up because I pushed the change from Mac and Mac/Android which I test on don't seem to mind about capitalisation so find the file anyway.

I'll update this evening and push an update

**Comment by realcopacetic** [comment]
Yep turns out Git doesn't automatically pick up case changes in file names! I've updated here - https://github.com/realcopacetic/skin.copacetic/commit/852ab523963c0a59dc0c68d71db1586ca7d2af8a

Will push to my repo this evening

**Comment by pepszzi** [comment]
Nice, that it was such a simple problem in the end. I added the files manually and now everything works again. 
I just love that skin :) 

Now the only question left is: why Kodi often crashes while browsing the settings? Does my log provided any insights into this?

**Comment by realcopacetic** [comment]
> Nice, that it was such a simple problem in the end. I added the files manually and now everything works again. I just love that skin :)
> 
> Now the only question left is: why Kodi often crashes while browsing the settings? Does my log provided any insights into this?

Nothing stands out. The settings window is actually a bunch of seperate windows

SkinSettings.xml --> Copacetic settings
SettingsCategory.xml --> Interface, Media, Player, System, Services, Games, PVR
SettingsProfile.xml --> Profiles
SettingsSystemInfo.xml --> System Information
FileManager.xml --> File Manager

Does the crashing always happen at the same time/place?

**Comment by pasnox** [comment]
no, it can happen any time when switching sections like skin settings, interface etc

**Comment by realcopacetic** [comment]
does it happen if you are switching between them quickly or slowly?

**Comment by pasnox** [comment]
quickly, i noticed that it would likely not crash when waiting some seconds before switching in a row

**Comment by pasnox** [comment]
hm it seems i can't crash it anymore with v1.1.27. Will test again in the following days.

**Comment by pepszzi** [comment]
> hm it seems i can't crash it anymore with v1.1.27. Will test again in the following days.

I can confirm that so far.
Thank you very much @realcopacetic

**Comment by pasnox** [comment]
unfortunately the crash issue is not fixed, while it's more hard recently to trigger it, it still occurs randomly 😭

**Comment by pepszzi** [comment]
Unfortunately I can confirm that. Same happens when you press the "player" icon on the OSD.

**Comment by pepszzi** [comment]
Setting the refresh rate to 50Hz seems to fixed this problem.

**Comment by pasnox** [comment]
I can make it crash easily that way

- do try a network stream that would fail 
- go in settings
- immediately push the down arrow
- crash

**Comment by pasnox** [comment]
Still reproducible, not easily though 😤

**Comment by realcopacetic** [comment]
I still can't replicate this on any of the platforms I have access to so it may be confined to libreelec. Has it always happened or since switching to omega? I find omega crashes more than nexus even with the same code as it's not as stable yet. 

The only thing I can think is it's some sort of CPU/ram spike caused by switching very quickly between windows as each different option at the top in settings is technically a different window. So it's having to switch mucj faster than most.skins where you back out of each section to a settings screen landing page, then choose a new window to open. I think arctic horizon 2 has a similar menu (that's where I got the idea) so might be worth seeing if it's happening with that skin.

The other thing I can do is make a test version of the skin/script that disables some of the background stuff happening in the script (monitoring settings / fetching backgrounds) to see if it's anything to do with that, bit I doubt it as that stuff is on a timed loop and this crash seems to happen when you do something

**Comment by realcopacetic** [comment]
> I can make it crash easily that way
> 
> - do try a network stream that would fail 
> - go in settings
> - immediately push the down arrow
> - crash

When you say network stream do you mean this happens when you're playing something ? Local or from online? If online what addons/sources are you using?

**Comment by realcopacetic** [comment]
> Unfortunately I can confirm that. Same happens when you press the "player" icon on the OSD.

I recently updated the osd, does this still happen? Is it during playback of streams or local content?

**Comment by pasnox** [comment]
Still reproducible. Rarely though

**Comment by realcopacetic** [comment]
hi @pasnox @pepszzi - can you let me know if this crash is still occurring in 1.5.9 (Nexus) / 2.5.9 (Omega) please, depending on which version you are on? I've justed pushed the update to my repo.

**Comment by realcopacetic** [comment]
hi @pasnox @pepszzi - is this crash still happening? I hope it's fixed but can't test

**Comment by pasnox** [comment]
still reproducible here, like before very randomly
